packages/firestore/src/local/simple_db.ts

```diff
@@ -294,23 +294,23 @@ export class SimpleDb {
         // caller.
         await transaction.completionPromise;
         return transactionFnResult;
-      } catch (e) {
+      } catch (error) {
         // TODO(schmidt-sebastian): We could probably be smarter about this and
         // not retry exceptions that are likely unrecoverable (such as quota
         // exceeded errors).
         const retryable =
           idempotent &&
-          isDomException(e) &&
+          error.name !== 'FirebaseError' &&
```

Comment:
Note that the instanceof check doesn't work here since the Async/Await wraps our FirestoreError in a generic Error.

Reply:
This might be worthwhile to add as an actual comment to the code rather than just the PR.

Reply:
Done